frontend-build-pipeline.yml

```diff
@@ -0,0 +1,52 @@
+trigger:
+- main
+- staging
+- development
+
+pool:
+  vmImage: ubuntu-latest
+
+steps:
+
+# Node.js is required to build Angular apps
+- task: NodeTool@0
+  inputs:
+    versionSpec: '12.14'
+  displayName: 'Install Node.js'
+
+# --dev option installs devDependencies from package.json
+# probably needed for running the Karma tests
+#
+# angular-devkit/build-angular
+# https://stackoverflow.com/questions/50333003/could-not-find-module-angular-devkit-build-angular
+- script: |
+    npm install --save-dev @angular-devkit/build-angular
+    npm install -g @angular/cli
+    npm install
+    ng build --prod
+  displayName: 'npm Install and Build'
+
+# continueOnError for now since unit tests fail
+- script: |
+    npx ng test --karma-config karma.conf.js --watch=false
```

Comment:
good work more or less figuring this part out. the `--karma-config` part is superfluous though since angular.json already tells it to read that file.

there are nicer ways to encapsulate running these commands with different arguments, both in the package.json as scripts, and in the angular.json as configurations, but that's more advanced and this is ok too.